src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -266,6 +267,12 @@
 
     }
 
+    private static boolean currentUserHasRunScriptsPermission() {
+        return Jenkins.getActiveInstance()
+                .getACL()
+                .hasPermission(Jenkins.RUN_SCRIPTS);
```

Comment:
This checks permissions on current user, which IIRC is `SYSTEM` when a build is automatically triggered (i.e. react to a webhook call). So I think this would be transitively by-passing the trusted flag for any user filing a PR.